core/domain/cron_services.py

```diff
@@ -56,6 +63,44 @@ def get_stuck_jobs(recency_msecs):
     return stuck_jobs
 
 
+def delete_models_marked_as_deleted():
+    """Hard-delete all models that are marked as deleted (have deleted field set
+    to True) and were last updated more than eight weeks ago.
+    """
+    date_now = datetime.datetime.utcnow()
+    date_before_which_to_hard_delete = (
+        date_now - PERIOD_TO_HARD_DELETE_MODELS_MARKED_AS_DELETED)
+    for model_class in models.Registry.get_all_storage_model_classes():
+        deleted_models = model_class.query(
+            model_class.deleted == True  # pylint: disable=singleton-comparison
+        ).fetch()
+        models_to_hard_delete = [
+            deleted_model for deleted_model in deleted_models
+            if deleted_model.last_updated < date_before_which_to_hard_delete
+        ]
+        if issubclass(model_class, base_models.VersionedModel):
+            model_class.delete_multi(
```

Comment:
IIRC, there was a limit on how many models you can delete in one call. If so we may have to delete in multiple calls.

Reply:
I'm not sure where I had seen this. But I'm assuming it won't be an issue!